Flaky TestJdbcVendorCompatibility - unable to download postgres docker image

```2021-09-29T15:39:04.2058393Z [ERROR] io.trino.jdbc.TestJdbcVendorCompatibility.setupServer  Time elapsed: 61.01 s  <<< FAILURE!
2021-09-29T15:39:04.2060263Z org.testcontainers.containers.ContainerLaunchException: Container startup failed
2021-09-29T15:39:04.2062139Z 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
2021-09-29T15:39:04.2064028Z 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:315)
2021-09-29T15:39:04.2065983Z 	at io.trino.jdbc.TestJdbcVendorCompatibility$PostgresqlReferenceDriver.<init>(TestJdbcVendorCompatibility.java:549)
2021-09-29T15:39:04.2068061Z 	at io.trino.jdbc.TestJdbcVendorCompatibility.setupServer(TestJdbcVendorCompatibility.java:91)
2021-09-29T15:39:04.2070251Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-09-29T15:39:04.2072174Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-09-29T15:39:04.2074397Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-09-29T15:39:04.2076094Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2021-09-29T15:39:04.2078443Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2021-09-29T15:39:04.2080329Z 	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
2021-09-29T15:39:04.2081999Z 	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:217)
2021-09-29T15:39:04.2083444Z 	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:144)
2021-09-29T15:39:04.2085485Z 	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:169)
2021-09-29T15:39:04.2087265Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
2021-09-29T15:39:04.2088820Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-09-29T15:39:04.2090427Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-09-29T15:39:04.2091532Z 	at java.base/java.lang.Thread.run(Thread.java:829)
2021-09-29T15:39:04.2093544Z Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=postgres:12.4, imagePullPolicy=DefaultPullPolicy())
2021-09-29T15:39:04.2095977Z 	at org.testcontainers.containers.GenericContainer.getDockerImageName(GenericContainer.java:1317)
2021-09-29T15:39:04.2098078Z 	at org.testcontainers.containers.GenericContainer.logger(GenericContainer.java:641)
2021-09-29T15:39:04.2099925Z 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:324)
2021-09-29T15:39:04.2101123Z 	... 16 more
2021-09-29T15:39:04.2103616Z Caused by: org.testcontainers.containers.ContainerFetchException: Failed to pull image: postgres:12.4
2021-09-29T15:39:04.2105300Z 	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:108)
2021-09-29T15:39:04.2106961Z 	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:28)
2021-09-29T15:39:04.2108619Z 	at org.testcontainers.utility.LazyFuture.getResolvedValue(LazyFuture.java:17)
2021-09-29T15:39:04.2109990Z 	at org.testcontainers.utility.LazyFuture.get(LazyFuture.java:39)
2021-09-29T15:39:04.2111688Z 	at org.testcontainers.containers.GenericContainer.getDockerImageName(GenericContainer.java:1315)
2021-09-29T15:39:04.2112954Z 	... 18 more
2021-09-29T15:39:04.2113445Z Caused by: java.lang.InterruptedException
2021-09-29T15:39:04.2115524Z 	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1040)
2021-09-29T15:39:04.2118922Z 	at java.base/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1345)
2021-09-29T15:39:04.2121185Z 	at java.base/java.util.concurrent.CountDownLatch.await(CountDownLatch.java:232)
2021-09-29T15:39:04.2123120Z 	at com.github.dockerjava.api.async.ResultCallbackTemplate.awaitCompletion(ResultCallbackTemplate.java:91)
2021-09-29T15:39:04.2126493Z 	at org.testcontainers.images.TimeLimitedLoggedPullImageResultCallback.awaitCompletion(TimeLimitedLoggedPullImageResultCallback.java:52)
2021-09-29T15:39:04.2129311Z 	at org.testcontainers.images.RemoteDockerImage.resolve(RemoteDockerImage.java:86)
2021-09-29T15:39:04.2130282Z 	... 22 more
2021-09-29T15:39:04.2132634Z 
2021-09-29T15:39:04.2135942Z [ERROR] io.trino.jdbc.TestJdbcVendorCompatibility.tearDown  Time elapsed: 0 s  <<< FAILURE!
2021-09-29T15:39:04.2137025Z java.lang.NullPointerException
2021-09-29T15:39:04.2138263Z 	at io.trino.jdbc.TestJdbcVendorCompatibility.tearDown(TestJdbcVendorCompatibility.java:119)
2021-09-29T15:39:04.2139977Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-09-29T15:39:04.2141901Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-09-29T15:39:04.2143945Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-09-29T15:39:04.2145475Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2021-09-29T15:39:04.2146883Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2021-09-29T15:39:04.2148596Z 	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
2021-09-29T15:39:04.2150102Z 	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:217)
2021-09-29T15:39:04.2151305Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:712)
2021-09-29T15:39:04.2152438Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
2021-09-29T15:39:04.2153657Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
2021-09-29T15:39:04.2155192Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
2021-09-29T15:39:04.2156693Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
2021-09-29T15:39:04.2158109Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-09-29T15:39:04.2159616Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-09-29T15:39:04.2160613Z 	at java.base/java.lang.Thread.run(Thread.java:829)
```

```
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=mongo:3.6.20, imagePullPolicy=DefaultPullPolicy())
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,6 +17,11 @@ env:
   MAVEN_FAST_INSTALL: "-B --strict-checksums -V --quiet -T C1 -DskipTests -Dair.check.skip-all"
   MAVEN_TEST: "-B --strict-checksums -Dair.check.skip-all --fail-at-end"
   RETRY: .github/bin/retry
+  # Testcontainers kills image pulls if they don't make progress for > 30s and retries for 2m before failing. This means
+  # that if an image doesn't download all it's layers within ~2m then any other concurrent pull will be killed because
+  # the Docker daemon only downloads 3 layers concurrently which prevents the other pull from making any progress.
+  # This value should be greater than the time taken for the longest image pull.
+  TESTCONTAINERS_PULL_PAUSE_TIMEOUT: 600
 
 # Cancel previous PR builds.
 concurrency:

